modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterShanpshotStreamerTest.java

```diff
@@ -0,0 +1,234 @@
+     */
+    @Test
+    public void testOtherCacheRestores() throws IgniteCheckedException {
+        String cname = "cache2";
+        String expectedWrn = U.field(DataStreamerUpdatesHandler.class, "WRN_MSG");
+
+        grid(0).createCache(new CacheConfiguration<>(dfltCacheCfg).setName(cname));
+
+        try (IgniteDataStreamer<Integer, Integer> ds = grid(0).dataStreamer(cname)) {
+            for (int i = 0; i < 100; ++i)
+                ds.addData(i, i);
+        }
+
+        AtomicBoolean stopLoad = new AtomicBoolean();
+
+        IgniteInternalFuture<?> loadFut = runLoad(grid(0), false, stopLoad);
+
+        try {
+            assertThrows(null, () -> snpMgr.createSnapshot(SNAPSHOT_NAME).get(), IgniteException.class, expectedWrn);
+        }
+        finally {
+            stopLoad.set(true);
+            loadFut.get();
+        }
+
+        grid(0).destroyCache(cname);
+        grid(0).destroyCache(dfltCacheCfg.getName());
+
+        snpMgr.restoreSnapshot(SNAPSHOT_NAME, Collections.singletonList(cname)).get();
+
+        for (int i = 0; i < 100; ++i)
+            assertEquals(i, grid(0).cache(cname).get(i));
+    }
+
+    /**
+     * Tests streaming into in-memory cache doesn't affect snapshot.
+     */
+    @Test
+    public void testStreamingIntoInMememoryDoesntAffectSnapshot() throws IgniteCheckedException {
+        String cache2Name = "cache2";
+        int loadCnt = 1000;
+
+        grid(0).createCache(new CacheConfiguration<>(dfltCacheCfg).setName(cache2Name));
+
+        try (IgniteDataStreamer<Object, Object> ds = grid(0).dataStreamer(cache2Name)) {
+            for (int i = 0; i < loadCnt; ++i)
+                ds.addData(i, i);
+        }
+
+        grid(0).destroyCache(dfltCacheCfg.getName());
+        dfltCacheCfg.setDataRegionName(INMEM_DATA_REGION);
+        dfltCacheCfg.setEncryptionEnabled(false);
+        grid(0).createCache(dfltCacheCfg);
+
+        AtomicBoolean stop = new AtomicBoolean();
+
+        IgniteInternalFuture<?> loadFut = runLoad(grid(2), false, stop);
+
+        try {
+            snpMgr.createSnapshot(SNAPSHOT_NAME).get();
+        }
+        finally {
+            stop.set(true);
+            loadFut.get();
+        }
+
+        grid(0).destroyCache(cache2Name);
+
+        snpMgr.restoreSnapshot(SNAPSHOT_NAME, null).get();
+
+        for (int i = 0; i < loadCnt; ++i)
+            assertEquals(i, grid(0).cache(cache2Name).get(i));
+    }
+
+    /**
+     * Tests snapshot process throws warning if required.
+     *
+     * @param mustFail If {@code true}, checks snapshot process produces the warning. Otherwise, checks snapshot
+     *                 process produces no warning.
+     * @param allowOverwrite 'allowOverwrite' setting.
+     */
+    private void doTestDataStreamerWhileSnapshot(boolean mustFail, boolean allowOverwrite) throws Exception {
+        String expectedWrn = U.field(DataStreamerUpdatesHandler.class, "WRN_MSG");
```

Comment:
Lets change a modifier of the `WRN_MSG` to a public.

Reply:
Fixed